wordpress-mobile/WordPress-iOS
# Add Site Name screen

Ref #18273 

This PR adds the Site Name screen, to be used in the new Site Creation flow.
This PR also contain some rearrangements of files and sub groups in the project group `Site Creation`

### Known issues
- This PR only implements the UI
- The UI is currently detached from the Site Creation flow, so it won't have the proper navigation buttons, nor the navigation behaviors
- The website title temporarily defaults to "Community & Non-Profit"
- The search bar placeholder temporarily defaults to "Search"

## To test:

<sub>Internal Test Plan: pc8HXX-fS-p2</sub>

### Pre-requisite
Since this new screen is not yet hooked up in the site creation flow, in order to test it, you might want to tweak the code somewhere, to instantiate and push it. For example, in `SiteIntentViewController` you could temporary change the code inside `tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath)` and replace the call to `selection(vertical)` (around line 177) with something like:
`navigationController?.pushViewController(SiteNameViewController(creator: SiteCreator()), animated: true)`

### Testing steps
- once implemented the pre-requisite, checkout, build and run this branch
- Select the entry point that you have chosen to show the Site Name screen UI
- Make sure the UI reflects the specs in #18273. More details in the screenshots and animation below:
- Make sure the "Continue" button is disabled when there is no text in the search field.

iPad portrait (dark) | iPad landscape (dark)
-- | --
<img height=400 src=https://user-images.githubusercontent.com/34376330/161649557-6e00645b-e1be-4efd-8a5d-2f15887fa3b5.PNG> | <img height=300 src=https://user-images.githubusercontent.com/34376330/161649598-aafe599b-0ba8-4739-9aaf-91c3b056fa56.PNG>


<p align=center>
<img height=450 src=https://user-images.githubusercontent.com/34376330/161649718-0058626a-96ba-4a9c-a4eb-b9307e9f7a04.mp4>
</p>
<p align=center>
<em>iPhone animation (light)</em>
</p>


## Regression Notes
1. Potential unintended areas of impact
None, this just adds a new screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
NA

3. What automated tests I added (or what prevented me from doing so)
None, this PR only deals with UI

PR submission checklist:

- [x] ~~I have considered adding accessibility improvements for my changes.~~ To be completed in a future PR.
